floss/get_go_strings.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+
+import sys
+import struct
+import logging
+import argparse
+from typing import List, Iterable, Optional
+
+import pefile
+
+from floss.results import StaticString, StringEncoding
+
+logger = logging.getLogger(__name__)
+
+MIN_STR_LEN = 4
+
+
+def extract_go_strings(
+    sample: str,
```

Comment:
general comment:
i think we should start prefering to use `pathlib.Path` for things that are filesystem paths. these objects have nice methods/accessors for common operations, and furthermore, make it clear the parameter contains a path, rather than some string information.

thoughts @mr-tz?
we can pull this into its own tracking issue. don't need to address in this PR.

Reply:
yes, these advantages sound great